### PR TITLE
LIVE-147: Functionality to play past event to all users

### PR DESCRIPTION
**Pull request by LVala**

### CODE REVIEW

**Review by Rados13 — CHANGES_REQUESTED** [review]
In addition, I think that we should add some tests regarding recordings endpoints. Also, CI fails.

File: assets/js/components/dashboard/EventsArea.tsx [hunk 1]
```diff
@@ -12,24 +12,31 @@ type EventsAreaProps = {
...
-  const listEvents = (upcoming: boolean) => {
+  const listEvents = (isRecording: boolean, upcoming: boolean) => {
     const curDate = new Date();
+    const events = isRecording ? recordings : webinars;
 
-    return webinars
-      .filter((elem) => (upcoming ? elem.startDate >= curDate : elem.startDate < curDate))
+    const filtered_events = events
+      .filter((elem) =>
+        isRecording ? true : upcoming ? elem.startDate >= curDate : elem.startDate < curDate
```
[the remark this answers is not in the document]
Comment (LVala) [reply assets/js/components/dashboard/EventsArea.tsx:24]: Fixed.

File: assets/js/components/dashboard/EventsArea.tsx [hunk 2]
```diff
@@ -12,24 +12,31 @@ type EventsAreaProps = {
...
   const toast = useToast();
 
-  const listEvents = (upcoming: boolean) => {
+  const listEvents = (isRecording: boolean, upcoming: boolean) => {
     const curDate = new Date();
+    const events = isRecording ? recordings : webinars;
 
-    return webinars
-      .filter((elem) => (upcoming ? elem.startDate >= curDate : elem.startDate < curDate))
+    const filtered_events = events
```
[the remark this answers is not in the document]
Comment (LVala) [reply assets/js/components/dashboard/EventsArea.tsx:22]: Fixed.

File: assets/js/components/event/ControlPanel.tsx [hunk 3]
```diff
@@ -228,7 +234,30 @@ const ControlPanel = ({
...
+                    onClick={() => {
+                      eventChannel.push("finish_event", {});
+                    }}
+                  >
+                    YES
+                  </button>
+                </div>
+              </PopoverBody>
+            </PopoverContent>
+          </Popover>
```
[the remark this answers is not in the document]
Comment (LVala) [reply assets/js/components/event/ControlPanel.tsx:260]: Fixed.

File: assets/js/utils/dashboardUtils.ts [hunk 4]
```diff
@@ -20,13 +20,15 @@ export const sendEventForm = async (
 ): Promise<void> => {
   const endpoint = "resources/webinars/" + uuid;
   const method = methodMap[modalType];
-
-  method(endpoint, { webinar: eventForm });
+  console.log("AAAAA");
```
Comment (Rados13) [inline assets/js/utils/dashboardUtils.ts:23]: Do we need this?

File: lib/membrane_live/event.ex [hunk 5]
```diff
@@ -216,26 +216,38 @@ defmodule MembraneLive.Event do
...
 
-      {:stop, :normal, state}
-    else
-      {peer_id, _peer_channel_id} =
-        state.peer_channels
-        |> Enum.find(fn {_peer_id, peer_channel_pid} -> peer_channel_pid == pid end)
+    cond do
+      pid == state.rtc_engine ->
+        event_span_id(state.event_id)
+        |> Membrane.OpenTelemetry.end_span()
```
Comment (Rados13) [inline lib/membrane_live/event.ex:234]:
```suggestion
        state.event_id
        |> event_span_id()
        |> Membrane.OpenTelemetry.end_span()
```

File: lib/membrane_live_web/channels/event_channel.ex [hunk 6]
```diff
@@ -52,12 +54,13 @@ defmodule MembraneLiveWeb.EventChannel do
...
              {:ok, email} <- Accounts.get_email(uuid),
+             is_moderator <- Webinars.check_is_user_moderator(uuid, id),
+             socket <- Socket.assign(socket, %{is_moderator: is_moderator, event_id: id}),
              [] <- Presence.get_by_key(socket, email),
-             {:ok, socket} <- create_event_stream(id, socket),
-             {:ok, is_presenter} = check_if_presenter(email, reloaded, id),
-             {:ok, is_request_presenting} = check_if_request_presenting(email, reloaded, id),
-             is_moderator <- Webinars.check_is_user_moderator(uuid, id) do
-          {:ok, socket} = if is_presenter, do: join_event_stream(socket), else: {:ok, socket}
+             {:ok, socket} <- create_event(socket, is_moderator),
```
Comment (Rados13) [inline lib/membrane_live_web/channels/event_channel.ex:60]: As now `is_moderator` is inside the socket the arrity of `create_event` can be changed to 1.

File: lib/membrane_live_web/channels/event_channel.ex [hunk 7]
```diff
@@ -98,12 +103,74 @@ defmodule MembraneLiveWeb.EventChannel do
...
+
+    if is_nil(socket.assigns.event_pid),
+      do: raise("Event was not started before first joining attempt")
+
+    peer_id = "#{UUID.uuid4()}"
+
+    send(socket.assigns.event_pid, {:add_peer_channel, self(), peer_id})
+
+    {:ok, Socket.assign(socket, %{peer_id: peer_id})}
+  end
```
Comment (Rados13) [inline lib/membrane_live_web/channels/event_channel.ex:157]: I think we should move all private functions to the end of the file.

File: assets/js/utils/dashboardUtils.ts (hunk 4 above)
Comment (LVala) [reply assets/js/utils/dashboardUtils.ts:23]: Fixed.

File: lib/membrane_live/event.ex (hunk 5 above)
Comment (LVala) [reply lib/membrane_live/event.ex:234]: Fixed.

File: lib/membrane_live_web/channels/event_channel.ex (hunk 6 above)
Comment (LVala) [reply lib/membrane_live_web/channels/event_channel.ex:60]: Fixed.

File: lib/membrane_live_web/channels/event_channel.ex (hunk 7 above)
Comment (LVala) [reply lib/membrane_live_web/channels/event_channel.ex:157]: Fixed.

**Review by Rados13 — APPROVED** [review]
Only fixing merge conflicts left.

**Pull request merged**